game-core/src/test/java/games/strategy/triplea/delegate/DiceRollTest.java

```diff
@@ -741,4 +750,61 @@ void testSbrRolls() {
     assertThat(StrategicBombingRaidBattle.getSbrRolls(bomber, british), is(2));
     assertThat(StrategicBombingRaidBattle.getSbrRolls(bomber, british), is(2));
   }
+
+  @Nested
+  @ExtendWith(MockitoExtension.class)
+  class SortAaHighToLowTest {
+
+    @Mock private Unit unit1;
+    @Mock private Unit unit2;
+    @Mock private Unit unit3;
+    @Mock private Unit unit4;
+    @Mock private Unit unit5;
+
+    private final Map<Unit, Tuple<Integer, Integer>> unitPowerAndRollsMap = new HashMap<>();
+    private final List<Unit> units = new ArrayList<>();
+
+    @BeforeEach
+    void setUp() {
+      units.addAll(List.of(unit1, unit2, unit3, unit4, unit5));
+      for (final var unit : units) {
+        final UnitType unitTypeMock = mock(UnitType.class);
+        final UnitAttachment unitAttachment = mock(UnitAttachment.class);
+        when(unitTypeMock.getAttachment(any())).thenReturn(unitAttachment);
+        when(unit.getType()).thenReturn(unitTypeMock);
+      }
+    }
+
+    @Test
+    void testAttacking() {
+      int index = 4;
+      for (final var unit : units) {
+        when(((UnitAttachment) unit.getType().getAttachment("")).getOffensiveAttackAa(any()))
+            .thenReturn(index / 2);
+        index--;
+      }
+      DiceRoll.sortAaHighToLow(units, gameData, false, unitPowerAndRollsMap);
+      assertThat(units.get(0), is(unit1));
+      assertThat(units.get(1), is(unit2));
+      assertThat(units.get(2), is(unit3));
+      assertThat(units.get(3), is(unit4));
+      assertThat(units.get(4), is(unit5));
+    }
+
+    @Test
+    void testDefending() {
```

Comment:
A javadoc comment on how this method is setup to test I think would go a long way. Looking at the loop and comparing to the last test case, it kinda seems like we would get a reverse sort order. Why do we expect the order that is being asserted?

Reply:
I'm pretty sure the sort algorithm used by java is stable, so because multiple values share the same value, they'll stay in their initial order